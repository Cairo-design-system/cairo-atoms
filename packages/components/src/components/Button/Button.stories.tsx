import React from 'react';
import { ComponentMeta, Story } from '@storybook/react';
import Button from './Button';

export default {
  title: 'Components/Button',
  component: Button,
} as ComponentMeta<any>;

export const Base: Story = (args: any) => <Button {...args} />
Base.args = {
  children: 'adaaoi'
}
