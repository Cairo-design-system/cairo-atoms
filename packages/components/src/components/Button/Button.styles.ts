import styled from "styled-components";

export const ButtonBase = styled.button`
    display: inline-flex;
    vertical-align: middle;
    align-items: center;
    justify-content: center;

    /* Safari button margins reset */
    /* See https://github.com/google/material-design-lite/issues/4008 */
    margin-top: 0px;
    margin-left: 0px;
`;