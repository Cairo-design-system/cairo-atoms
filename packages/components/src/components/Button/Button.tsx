import React, { useContext, useState } from 'react';
import { CodeOutlined } from '@escoleme/meicons-react';

import { FilledButton, LinkFilledButton } from './styles/Button.filled.styles';
import { FilledTonalButton, LinkFilledTonalButton } from './styles/Button.tonal.styles';
import { LinkOutlinedButton, OutlinedButton } from './styles/Button.outlined.styles';
import { LinkTextButton, TextButton } from './styles/Button.text.styles';

import { automationAttribute } from '../../helpers/utils';
import { ButtonProps } from './Button.types';
import { ThemeContext } from 'styled-components';


const Button = React.forwardRef(
  (
    {
      iconAlign = 'left',
      variant = 'filled',
      size ='medium',
      ...props
    }: ButtonProps,
    ref: ButtonProps['ref']
  ) => {

  const { comp } = useContext(ThemeContext);
  const button = comp.button;

  const buttonStyles = button[variant];
  const { iconSize } = buttonStyles.layout[size] ? buttonStyles.layout[size] : buttonStyles.layout.medium;

  const content: JSX.Element[] = [];
  const icon = props.icon;

  const iconNode = icon ? icon : null;

  const handleClick = (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement, MouseEvent>) => {
    const { onClick, disabled, disabledOnClick, loading } = props;

    if(disabledOnClick || disabled || loading) {
      e.preventDefault();
      return;
    }

    (onClick as React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>)?.(e);
  }


  // Icone de carregando
  if (props.loading) {
    content.push(<CodeOutlined spin style={{ fontSize: iconSize }} />);
  }

  // Botão com ícone à esquerda
  if (iconNode && iconAlign === "left") {
    content.push(iconNode);
  }

  // Conteúdo do botão
  if (props.label) {

    // Se estiver carregando e tiver uma LoadingLabel
    if (props.loading && props.loadingLabel) {
      content.push(<>{props.loadingLabel}</>);

    // Conteúdo normal
    } else {
      content.push(<span style={{
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis'
      }}>{props.label}</span>);
    }
  }

  // Botão com ícone à direita
  if (iconNode && iconAlign === "right") {
    content.push(iconNode);
  }

  // Se tiver a propriedade href então o botão vai ser um a senão vai ser um button.
  let ButtonStyle = FilledButton;

  if (variant === 'filled') {
    ButtonStyle = props.href ? LinkFilledButton : FilledButton;
  } else if (variant === "tonal") {
    ButtonStyle = props.href ? LinkFilledTonalButton : FilledTonalButton;
  } else if (variant === "outlined") {
    ButtonStyle = props.href ? LinkOutlinedButton : OutlinedButton;
  } else if (variant === "text") {
    ButtonStyle = props.href ? LinkTextButton : TextButton;
  }

  if(props.rtl) {
    content.reverse()
  }

  return (
    <ButtonStyle
      {...automationAttribute("button")}

      {...props}
      onClick={handleClick}
      iconAlign={iconAlign}
      size={size}
      loading={props.loading}
      ref={ref}
    >
      {content}
    </ButtonStyle>
  );
});

Button.displayName = 'Button';

export default Button;
