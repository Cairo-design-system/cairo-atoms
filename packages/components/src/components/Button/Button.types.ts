import React from 'react';
import { tupleString } from '../../helpers/utils';

const style = tupleString('filled', 'tonal', 'outlined', 'text');
export type Style = typeof style[number];

// const ButtonShapes = tupleString('default', 'circle', 'round');
// export type ButtonShape = typeof ButtonShapes[number];

const ButtonHTMLTypes = tupleString('button', 'submit', 'reset');
export type ButtonHTMLType = (typeof ButtonHTMLTypes)[number];

const ButtonTargets = tupleString('_black', '_self', '_parent', "_top");
export type ButtonTarget = (typeof ButtonTargets)[number];

const ButtonIconAligns = tupleString('left', 'right');
export type ButtonIconAlign = (typeof ButtonIconAligns)[number];

export interface IButtonProps {
    id?: string;
    className?: string;
    variant?: Style;
    href?: string;
    type?: ButtonHTMLType;
    target?: ButtonTarget;
    disabled?: boolean;
    block?: boolean;
    onClick?: React.MouseEventHandler<HTMLElement>;
    loading?: boolean;
    children?: React.ReactNode;
    success?: boolean;
    danger?: boolean;
    onColor?: boolean;
    'data-testid'?: string;
    /**
     The ref to the HTML DOM element
     */
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    ref?: React.Ref<any>;



    
    size?: any;


    

    icon?: React.ReactNode;
    /** Align Icon Left or Right */
    iconAlign?: ButtonIconAlign;
    rel?: string; // TODO: use `noopener` as default
    loadingLabel?: string;
}
