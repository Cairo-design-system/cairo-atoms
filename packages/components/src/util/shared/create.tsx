import { render, RenderFn } from "./render";
import { RenderResult } from '@testing-library/react';

const create = (
    ...args: Parameters<RenderFn<RenderResult>>
): ChildNode | HTMLCollection | null => {

    const { container } = render(...args);

    return container.children.length > 1
      ? container.children
      : container.firstChild;
};
  
export { create };