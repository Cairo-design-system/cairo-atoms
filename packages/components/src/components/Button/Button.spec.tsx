import React, { createRef } from 'react';

import Button from './Button';
import { ButtonProps } from './Button.types';

import { render, RenderFn } from '../../util/shared/render';
import { renderToHtml } from '../../util/shared/renderToHtml';
import { create } from '../../util/shared/create';
import { axe } from '../../util/shared';
import { CodeFilled } from '@escoleme/meicons-react';
import { act, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Button', () => {

  function renderButton<T>(renderFn: RenderFn<T>, props: ButtonProps) {
    return renderFn(<Button {...props} />);
  }

  const baseProps = { label: 'Button' };

  it('mount correctly', () => {
    const props: ButtonProps = {
      ...baseProps,
      disabled: true
    };
    const { unmount } = renderButton(render, props);
    
    expect(() => {
      unmount();
    }).not.toThrow();
  });

  describe('variants', () => {
    describe('Filled', () => {
      it('should render a filled button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          variant: 'filled',
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a danger button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          danger: true,
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a success button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          success: true,
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a disabled button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          disabled: true,
        });
        expect(wrapper).toMatchSnapshot();
      });
    })
    
    describe('Filled tonal', () => {
      it('should render a filled tonal button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          variant: 'tonal',
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a danger button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          danger: true,
          variant: 'tonal'
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a success button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          success: true,
          variant: 'tonal'
        });
        expect(wrapper).toMatchSnapshot();
      });
    })
    
    describe('Outlined', () => {
      it('should render a outlined button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          variant: 'outlined',
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a danger button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          danger: true,
          variant: 'outlined'
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a success button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          success: true,
          variant: 'outlined'
        });
        expect(wrapper).toMatchSnapshot();
      });
    })
    
    describe('Text', () => {
      it('should render a text button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          variant: 'text',
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a danger button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          danger: true,
          variant: 'text'
        });
        expect(wrapper).toMatchSnapshot();
      });
  
      it('should render a success button', () => {
        const wrapper = renderButton(create, {
          ...baseProps,
          success: true,
          variant: 'text'
        });
        expect(wrapper).toMatchSnapshot();
      });
    })
  });

  describe('styles', () => {
    it('should render a small button', () => {
      const wrapper = renderButton(create, {
        ...baseProps,
        size: "small",
      });
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a medium button', () => {
      const wrapper = renderButton(create, {
        ...baseProps,
        size: "medium",
      });
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a large button', () => {
      const wrapper = renderButton(create, {
        ...baseProps,
        size: "large",
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('loading', () => {
    it('should render a button with loading styles', () => {
      const wrapper = renderButton(create, {
          ...baseProps,
          loading: true,
          loadingLabel: 'Loading',
      });
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a button with loading styles and only icon', () => {
      const wrapper = renderButton(create, {
          loading: true,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('icon', () => {
    it('should render a button with left icon', () => {
      const wrapper = renderButton(create, {
          ...baseProps,
          icon: <CodeFilled style={{ fontSize: 18 }} />
      });

      expect(wrapper).toMatchSnapshot();
    });

    it('should render a button with right icon', () => {
      const wrapper = renderButton(create, {
          ...baseProps,
          icon: <CodeFilled style={{ fontSize: 18 }} />,
          iconAlign: 'right'
      });

      expect(wrapper).toMatchSnapshot();
    });

    it('should render a button with rtl', () => {
      const wrapper = renderButton(create, {
          ...baseProps,
          icon: <CodeFilled style={{ fontSize: 18 }} />,
          rtl: true
      });
      
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a button with only an icon', () => {
      const wrapper = renderButton(create, {
          icon: <CodeFilled style={{ fontSize: 18 }} />,
      });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('disabled', () => {
    it('should not call the onClick handler when clicked', () => {
      const props: ButtonProps = { 
        ...baseProps,
        onClick: jest.fn(),
        disabled: true,
        variant: 'text',
        'data-testid': 'button'
      };
      const { getByTestId } = renderButton(render, props);

      const button = getByTestId('button');

      fireEvent.click(button)

      expect(props.onClick).toHaveBeenCalledTimes(0);
    });
  });

  describe('business logic', () => {
    it('should render as a link when passed the href prop', () => {
      const props = {
        ...baseProps,
        href: '#',
        onClick: jest.fn(),
        'data-testid': 'button'
      };

      const { getByTestId } = renderButton(render, props);
      const button = getByTestId('button');
      expect(button.tagName).toBe('A');
      expect(button).toHaveAttribute('href');
    });

    it('should render loading button with loading label', () => {
      const loadingLabel = 'Loading';
      const props = {
        ...baseProps,
        loading: true,
        loadingLabel,
      };

      const { getByText } = renderButton(render, props);
      expect(getByText(loadingLabel)).toBeVisible();
    });

    it('should call the onClick handler when clicked', async () => {
      const props = {
        ...baseProps,
        onClick: jest.fn(),
        'data-testid': 'button',
      };
      const { getByTestId } = renderButton(render, props);

      act(() => {
        userEvent.click(getByTestId('button'));
      });

      expect(props.onClick).toHaveBeenCalledTimes(1);
    });

    it('should not call the onClick handler when clicked with the disabledOnClick prop activated', async () => {
      const props = {
        ...baseProps,
        onClick: jest.fn(),
        'data-testid': 'button',
        disabledOnClick: true
      };
      const { getByTestId } = renderButton(render, props);

      act(() => {
        userEvent.click(getByTestId('button'));
      });

      expect(props.onClick).toHaveBeenCalledTimes(0);
    });

    it('should accept a working ref for a button', () => {
      const tref = createRef<any>();
      const { container } = render(
        <Button ref={tref} label="This is a button" />,
      );
      const button = container.querySelector('button');
      expect(tref.current).toBe(button);
    });

    it('should accept a working ref for a link', () => {
      const tref = createRef<any>();
      const { container } = render(
        <Button href="http://sumup.com" ref={tref} label="This is a button" />,
      );
      const button = container.querySelector('a');
      expect(tref.current).toBe(button);
    });
  });

  describe('link', () => {
    it('should render as a link when passed the href prop', () => {
      const props = { 
        ...baseProps,
        href: "#",
        onClick: jest.fn(),
        disabled: true,
        'data-testid': 'button'
      };
      const { getByTestId } = renderButton(render, props);

      const button = getByTestId('button');

      fireEvent.click(button)

      expect(props.onClick).toHaveBeenCalledTimes(0);
    });

    it('should render as a link when passed the href prop', () => {
      const props: ButtonProps = { 
        ...baseProps,
        href: "#",
        onClick: jest.fn(),
        disabled: true,
        variant: 'tonal',
        'data-testid': 'button'
      };
      const { getByTestId } = renderButton(render, props);

      const button = getByTestId('button');

      fireEvent.click(button)

      expect(props.onClick).toHaveBeenCalledTimes(0);
    });
    
    it('should render as a link when passed the href prop', () => {
      const props: ButtonProps = { 
        ...baseProps,
        href: "#",
        onClick: jest.fn(),
        disabled: true,
        variant: 'outlined',
        'data-testid': 'button'
      };
      const { getByTestId } = renderButton(render, props);

      const button = getByTestId('button');

      fireEvent.click(button)

      expect(props.onClick).toHaveBeenCalledTimes(0);
    });
    
    it('should render as a link when passed the href prop', () => {
      const props: ButtonProps = { 
        ...baseProps,
        href: "#",
        onClick: jest.fn(),
        disabled: true,
        variant: 'text',
        'data-testid': 'button'
      };
      const { getByTestId } = renderButton(render, props);

      const button = getByTestId('button');

      fireEvent.click(button)

      expect(props.onClick).toHaveBeenCalledTimes(0);
    });
  })

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(<Button label="Button" />);
      const actual = await axe(wrapper);
      expect(actual).toHaveNoViolations();
    });

    it('should meet accessibility guidelines for Loading button', async () => {
      const wrapper = renderToHtml(
        <Button loading loadingLabel="Loading" />,
      );

      const actual = await axe(wrapper);
      expect(actual).toHaveNoViolations();
    });
  });
})