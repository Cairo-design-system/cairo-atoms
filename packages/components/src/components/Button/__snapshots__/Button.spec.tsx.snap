// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button icon should render a button with left icon 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 hgXHin ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    aria-label="code"
    class="anticon anticon-code"
    role="img"
    style="font-size: 18px;"
  >
    <svg
      aria-hidden="true"
      data-icon="code"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM513.1 518.1l-192 161c-5.2 4.4-13.1.7-13.1-6.1v-62.7c0-2.3 1.1-4.6 2.9-6.1L420.7 512l-109.8-92.2a7.63 7.63 0 01-2.9-6.1V351c0-6.8 7.9-10.5 13.1-6.1l192 160.9c3.9 3.2 3.9 9.1 0 12.3zM716 673c0 4.4-3.4 8-7.5 8h-185c-4.1 0-7.5-3.6-7.5-8v-48c0-4.4 3.4-8 7.5-8h185c4.1 0 7.5 3.6 7.5 8v48z"
      />
    </svg>
  </span>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button icon should render a button with only an icon 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 cvTHPJ ceWXkd"
  data-medesign-key="button"
>
  <span
    aria-label="code"
    class="anticon anticon-code"
    role="img"
    style="font-size: 18px;"
  >
    <svg
      aria-hidden="true"
      data-icon="code"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM513.1 518.1l-192 161c-5.2 4.4-13.1.7-13.1-6.1v-62.7c0-2.3 1.1-4.6 2.9-6.1L420.7 512l-109.8-92.2a7.63 7.63 0 01-2.9-6.1V351c0-6.8 7.9-10.5 13.1-6.1l192 160.9c3.9 3.2 3.9 9.1 0 12.3zM716 673c0 4.4-3.4 8-7.5 8h-185c-4.1 0-7.5-3.6-7.5-8v-48c0-4.4 3.4-8 7.5-8h185c4.1 0 7.5 3.6 7.5 8v48z"
      />
    </svg>
  </span>
</button>
`;

exports[`Button icon should render a button with right icon 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 kXQcXI ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
  <span
    aria-label="code"
    class="anticon anticon-code"
    role="img"
    style="font-size: 18px;"
  >
    <svg
      aria-hidden="true"
      data-icon="code"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM513.1 518.1l-192 161c-5.2 4.4-13.1.7-13.1-6.1v-62.7c0-2.3 1.1-4.6 2.9-6.1L420.7 512l-109.8-92.2a7.63 7.63 0 01-2.9-6.1V351c0-6.8 7.9-10.5 13.1-6.1l192 160.9c3.9 3.2 3.9 9.1 0 12.3zM716 673c0 4.4-3.4 8-7.5 8h-185c-4.1 0-7.5-3.6-7.5-8v-48c0-4.4 3.4-8 7.5-8h185c4.1 0 7.5 3.6 7.5 8v48z"
      />
    </svg>
  </span>
</button>
`;

exports[`Button icon should render a button with rtl 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 kXQcXI ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
  <span
    aria-label="code"
    class="anticon anticon-code"
    role="img"
    style="font-size: 18px;"
  >
    <svg
      aria-hidden="true"
      data-icon="code"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM513.1 518.1l-192 161c-5.2 4.4-13.1.7-13.1-6.1v-62.7c0-2.3 1.1-4.6 2.9-6.1L420.7 512l-109.8-92.2a7.63 7.63 0 01-2.9-6.1V351c0-6.8 7.9-10.5 13.1-6.1l192 160.9c3.9 3.2 3.9 9.1 0 12.3zM716 673c0 4.4-3.4 8-7.5 8h-185c-4.1 0-7.5-3.6-7.5-8v-48c0-4.4 3.4-8 7.5-8h185c4.1 0 7.5 3.6 7.5 8v48z"
      />
    </svg>
  </span>
</button>
`;

exports[`Button loading should render a button with loading styles 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 eanouO ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    aria-label="code"
    class="anticon anticon-code anticon-spin"
    role="img"
    style="font-size: 18px;"
  >
    <svg
      aria-hidden="true"
      data-icon="code"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
      />
    </svg>
  </span>
  Loading
</button>
`;

exports[`Button loading should render a button with loading styles and only icon 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 fMBosi ceWXkd"
  data-medesign-key="button"
>
  <span
    aria-label="code"
    class="anticon anticon-code anticon-spin"
    role="img"
    style="font-size: 18px;"
  >
    <svg
      aria-hidden="true"
      data-icon="code"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
      />
    </svg>
  </span>
</button>
`;

exports[`Button styles should render a large button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 gAsQie ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button styles should render a medium button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 bezjLz ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button styles should render a small button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 gLPlXh ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Filled should render a danger button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 bezjLz jFDgWw"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Filled should render a disabled button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 kEFyyi hBZVRu"
  data-medesign-key="button"
  disabled=""
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Filled should render a filled button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 bezjLz ceWXkd"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Filled should render a success button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonfilledstyles__FilledButton-sc-1s1d10f-0 bezjLz eyThUo"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Filled tonal should render a danger button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttontonalstyles__FilledTonalButton-sc-1xh321n-0 bezjLz gYJSSE"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Filled tonal should render a filled tonal button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttontonalstyles__FilledTonalButton-sc-1xh321n-0 bezjLz llaODQ"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Filled tonal should render a success button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttontonalstyles__FilledTonalButton-sc-1xh321n-0 bezjLz caRwxc"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Outlined should render a danger button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonoutlinedstyles__OutlinedButton-sc-1daesx2-0 bezjLz kQunRG"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Outlined should render a outlined button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonoutlinedstyles__OutlinedButton-sc-1daesx2-0 bezjLz hHjZSY"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Outlined should render a success button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttonoutlinedstyles__OutlinedButton-sc-1daesx2-0 bezjLz doNDOT"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Text should render a danger button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttontextstyles__TextButton-sc-1oc6mlh-0 bezjLz hGRjNb"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Text should render a success button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttontextstyles__TextButton-sc-1oc6mlh-0 bezjLz cVQWFG"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;

exports[`Button variants Text should render a text button 1`] = `
<button
  class="Buttonbasestyles__ButtonBase-sc-ybojtw-0 Buttontextstyles__TextButton-sc-1oc6mlh-0 bezjLz jmYcTn"
  data-medesign-key="button"
  label="Button"
>
  <span
    style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Button
  </span>
</button>
`;
